chore: Allow partial configuration file

PThorpe92 opened the pull request:
## Description
This PR allows for the mandatory args that need to be generated from the config file, if the config file is found, be filled in with default arguments should they not be included in the found config file.

Please let me know if I missed anything, and thank you for the awesome program :+1: 

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
closes #105 

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
I tested this locally on my machine by placing a config file in ~/.config/gpg-tui and 
filling adding and removing parts to make sure the program will still start.
<!--- Include details of your testing environment, and the tests you ran to -->
I also wrote a test `test_args_partial_config` and included a partial conifg file `gpg-tui-part` in the config directory of the repo, and ran assertions for the missing arguments.

## Output (if appropriate):
<!--- What is the expected output of the project after these changes? -->
The program will still start, despite fields missing in the configuration file.
<!--- You might use logs, screenshots or files depending on the behaviour. --->
PREV:
![image](https://github.com/orhun/gpg-tui/assets/121899304/e057e720-9f52-495a-9199-c49aba0b06c1)
AFTER:
Well it works :+1: a screenshot of the program running is probably overkill.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (no code change)
- [ ] Refactor (refactoring production code)
- [ ] Other <!--- (provide information) -->

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] I have formatted the code with [rustfmt](https://github.com/rust-lang/rustfmt).
- [x] I checked the lints with [clippy](https://github.com/rust-lang/rust-clippy).
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

orhun commented on src/config.rs:
```diff
@@ -26,15 +26,15 @@ pub struct Config {
 #[derive(Debug, Default, Serialize, Deserialize)]
 pub struct GeneralConfig {
```
Any chance that we can make the sections (`[general]` and `[gpg]`) optional as well?

orhun commented on src/config.rs:
```diff
@@ -163,20 +163,43 @@ impl Config {
 	pub fn update_args(&self, mut args: Args) -> Args {
-		args.armor = self.gpg.armor;
-		args.splash = self.general.splash;
+		let default_color = Color::from("gray");
+		let default_style = Style::from_str("plain", true).unwrap_or_default();
+		let default_file_exp = String::from("xplr");
```
Can you use constants for these values?

PThorpe92 replied:
Like do you want lazy_static/and get those Color + Style variables themselves made constant? or do you want just the &str values made constant and put into the runtime functions there?

PThorpe92 replied:
I would assume the latter actually now that I think about it.

orhun reviewed (changes requested):
Thanks for picking this up! 💖 

I just had some comments for improvement.

PThorpe92 commented on config/gpg-tui-part.toml:
```diff
@@ -0,0 +1,27 @@
+[general]
```
Yeah for sure I do agree about it being confusing, I was just semi-unsure about how to write proper tests for something like this. Normally that file may have been included in a tests/ type directory. How would you like me to go about the test? Maybe create a temp file, write
```
[general]
splash = true
```
to it, and then run the test on that?

orhun replied:
Yeah I think that approach is better. You can possibly write it to a temporary directory as well.

PThorpe92 commented:
Take a look and let me know what you think, by making those both Option<T>, it did add some complexity when setting the default values but I added the two tests, one for each section, with the opposing one missing and then tested in both cases for each one.

I also fixed the unrelated clippy warnings that somehow still made the CI fail :eyes:

orhun commented on Cargo.lock:
Are the changes in `Cargo.lock` necessary?

PThorpe92 replied:
I am honestly not sure how/why this would have been generated, as I did not add/change any dependencies. Very strange.

orhun replied:
Ah, it might be because I updated some dependencies in the master branch. Let's ignore it.

orhun commented on src/config.rs:
```diff
@@ -163,49 +174,131 @@ impl Config {
+			assert_eq!(args.armor, false);
+			assert_eq!(args.default_key, None);
+		}
+		temp_file.flush()?;
```
What's the benefit of using `flush` here? Is it really needed?

PThorpe92 replied:
Ah, you are correct as I forgot to remove that when I changed from write to write_all

orhun reviewed (commented):
Hey, overall it looks good. I just pushed e89b2be18b4ad3dd12d9ac356ef0a8a4f99f5d41 for cleanup.

Also I have 2 questions, then we can merge 🐻

orhun reviewed (approved):
Thanks for your contribution!
